heron/schedulers/src/java/com/twitter/heron/scheduler/nomad/NomadScheduler.java

```diff
@@ -310,8 +310,10 @@ Task getTask(String taskName, int containerIndex,
         throw new RuntimeException(msg, e);
       }
       Map<String, String> metaData = jobActual.getMeta();
-      if (metaData.get(NomadConstants.NOMAD_TOPOLOGY_NAME).equals(topologyName)) {
-        ret.add(jobActual);
+      if (metaData != null && metaData.containsKey(NomadConstants.NOMAD_TOPOLOGY_NAME)) {
```

Comment:
Maybe a stupid question, can we use if (metaData != null && metaData.get(NomadConstants.NOMAD_TOPOLOGY_NAME) == topologyName) here?

Also if we need to break it into two if statements, I think if (metaData != null ) { if (...) } might be better.

Reply:
We cannot do "metaData.get(NomadConstants.NOMAD_TOPOLOGY_NAME) == topologyName)" because first in java to compare strings you have to your .equals instead of "==".  "==" will check if its the same instance rather if it has to same value.  If you use .equals we need to make sure that its not null or NPE will be thrown.